### PR TITLE
Book an apppointment: API documentation

### PR DESCRIPTION
**Pull request by Debas-31**
- [x]  Solved issue #10, added API documentation for the controllers and their actions

### CODE REVIEW

File: spec/requests/api/motorcycles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        run_test!
+      end
+    end
+
+    post('create motorcycle') do
+      response(200, 'successful') do
+        after do |example|
+          example.metadata[:response][:content] = {
+            'application/json' => {
+              example: JSON.parse(response.body, symbolize_names: true)
```
Comment (Achilles-Dev) [inline spec/requests/api/motorcycles_spec.rb]: Please add the user_id parameter for the Motorcycles post method. Also, add the schema properties for the post method.

example:
```
parameter name: 'user_id', in: :query, type: :integer, description: 'user_id', schema: {
        type: :object,
        properties: {
          motors: {
            model: { type: :string },
            image: { type: :string },
            price: { type: :integer },
            description: { type: :text },
            duration_months: { type: :integer }
          }
        }
      }
```

File: spec/requests/api/reservations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        after do |example|
+          example.metadata[:response][:content] = {
+            'application/json' => {
+              example: JSON.parse(response.body, symbolize_names: true)
+            }
+          }
+        end
+        run_test!
+      end
+    end
```
Comment (Achilles-Dev) [inline spec/requests/api/reservations_spec.rb]: Please add the user_id parameter for the Motorcycles post method. Also, add the schema properties for the post method.

example:
```
parameter name: motor_id, in: :query, type: :integer,  schema: {
        type: :object,
        properties: {
          reservations: {
            city: { type: :string },
            date: { type: :string}
          }
        }
      }
```

File: spec/requests/api/reservations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
+require 'swagger_helper'
+
+RSpec.describe 'api/reservations', type: :request do
+  path '/api/users/{user_id}/reservations' do
```
Comment (Achilles-Dev) [inline spec/requests/api/reservations_spec.rb]: Please change the path to reservations to `'api/reservations'`.
The `user_id` in the path is a query parameter and not a path parameter.

File: spec/requests/api/reservations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    # You'll want to customize the parameter types...
+    parameter name: 'user_id', in: :path, type: :string, description: 'user_id'
```
Comment (Achilles-Dev) [inline spec/requests/api/reservations_spec.rb]: The `user_id` should be a query parameter so change `path` to `query`.

File: spec/requests/api/reservations_spec.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+              example: JSON.parse(response.body, symbolize_names: true)
+            }
+          }
+        end
+        run_test!
+      end
+    end
+  end
+
+  path '/api/users/{user_id}/reservations/{id}' do
```
Comment (Achilles-Dev) [inline spec/requests/api/reservations_spec.rb]: Please change the path to `'api/reservations/{id}'`.
The path does not require a `user_id`.
Thank you.

**Review by Achilles-Dev — CHANGES_REQUESTED** [review]
Hi @Debas-31 :wave:

Good job implementing the API documentation.
You still need to work on some issues to go to the next step, but you are almost there!

## Required Changes ♻️
- [ ] Please add the paths for `login`/ `sign_in` and `register`/`sign_up` by a user.

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

File: spec/requests/api/reservations_spec.rb (hunk 5 above)
Comment (Debas-31) [reply spec/requests/api/reservations_spec.rb]: Thank you for the feedback; I will implement the required changes for the motorcycle and reservation integration test.

**Review by Achilles-Dev — APPROVED** [review]
Hi @Debas-31,

👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ ✔️ ✔️ 

👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

_Check for optional comments below_

Cheers and Happy coding!👏👏👏

**Pull request merged**